libr/util/strbuf.c

```diff
@@ -293,16 +311,25 @@ R_API ut8 *r_strbuf_getbin(RStrBuf *sb, int *len) {
 	return (ut8 *)(sb->ptr ? sb->ptr : sb->buf);
 }
 
+// TODO: expose aa r_strbuf_getdup() ?
+static inline char *get_strdup(RStrBuf *sb) {
+	return sb->ptr
+		? sb->ro
+			? r_mem_dup (sb->ptr, sb->len)
```

Comment:
If the weak pointer is a null-terminated string then you would get a non-null terminated one from here, right? Probably better to use ptrlen.

Reply:
weak have nothing to do with null or non null terminated strings. thats completely unrelated.

for simplicity we can nullterminate all the things

Reply:
Yes it has something to do with it here. If you currently do
```
r_strbuf_setptr(sb, "fuck", -1);
char *s = r_strbuf_drain(sb);
```
then s will not be null-terminated.